cmd/protoc-gen-gorums/dev/strictordering.go

```diff
@@ -0,0 +1,165 @@
+package dev
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"sync"
+	atomic "sync/atomic"
+	"time"
+
+	"github.com/relab/gorums"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/codes"
+	status "google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type strictOrderingResult struct {
+	nid   uint32
+	reply *anypb.Any
+	err   error
+}
+
+type requestHandler func(*gorums.Message) *gorums.Message
+
+type strictOrderingManager struct {
+	msgID    uint64
+	recvQ    map[uint64]chan *strictOrderingResult
```

Comment:
Consider if perhaps [`sync.Map`](https://golang.org/pkg/sync/#Map) should be used instead, because:
_The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex._

It would be interesting to do a performance comparison, given the comment from the documentation, which seems to match our use case, since we only ever write to the same key once, but read it many times.